### PR TITLE
fix lazy tensor detach and clone

### PR DESCRIPTION
**Pull request by hjchen2**

### CODE REVIEW

File: oneflow/core/framework/tensor.cpp [hunk 1]
```diff
@@ -91,6 +102,18 @@ Maybe<Tensor> MirroredTensor::clone() const {
   return JUST(functional::Copy(input, device_type, device_id, /*pin_memory=*/pin_memory));
 }
 
+Maybe<void> MirroredTensor::set_data(const std::shared_ptr<Tensor>& other) {
+  CHECK_OR_RETURN(this->is_leaf()) << "Can only set leaf tensor's data.";
+  const auto& mirrored_tensor = std::dynamic_pointer_cast<MirroredTensor>(JUST(other->detach()));
+  CHECK_NOTNULL_OR_RETURN(mirrored_tensor);
```
Comment (wyg1997) [inline oneflow/core/framework/tensor.cpp:108]: 这里可能要加错误提示了

**Review by wyg1997 — APPROVED** [review]
test_graph_clone 要不要改成 test_graph_tensor_clone，因为 clone 和 detach 是 Tensor 的方法，test_graph_clone.py 不太准确，或者合一起叫 test_graph_tensor_method.py ？

File: oneflow/core/framework/tensor.cpp (hunk 1 above)
Comment (hjchen2) [reply oneflow/core/framework/tensor.cpp:108]: > test_graph_clone 要不要改成 test_graph_tensor_clone，因为 clone 和 detach 是 Tensor 的方法，test_graph_clone.py 不太准确，或者合一起叫 test_graph_tensor_methods.py/test_graph_tensor.py ？

嗯，确实不大准确，我来改一下

File: oneflow/core/framework/tensor.cpp [hunk 2]
```diff
@@ -136,6 +160,7 @@ bool ConsistentTensor::is_cuda() const {
 
 Maybe<Tensor> ConsistentTensor::detach() const {
   std::shared_ptr<Tensor> tensor = std::make_shared<ConsistentTensor>(JUST(impl_->detach()));
+  if (this->is_lazy()) { JUST(tensor->BorrowTensorName(this)); }
```
Comment (strint) [inline oneflow/core/framework/tensor.cpp:163]: 复用名字，表示下游消费者消费的还是原 op 的输出，数值和内存是共享的。

在后向计算的逻辑是特别的，这里要做到中断 backward ？

Comment (hjchen2) [reply oneflow/core/framework/tensor.cpp:163]: 中断是通过requires_grad来实现的，不是通过name复用，name复用在构图上只是加一条连边

**Review by strint — APPROVED** [review]
